homeassistant/components/tesla_wall_connector/config_flow.py

```diff
@@ -0,0 +1,160 @@
+                self.ip_address,
+                ex,
+            )
+            return self.async_abort(reason="cannot_connect")
+
+        self.serial_number = version.serial_number
+
+        await self.async_set_unique_id(self.serial_number)
+        self._abort_if_unique_id_configured(updates={CONF_HOST: self.ip_address})
+
+        _LOGGER.debug(
+            "No entry found for wall connector with IP %s. Serial nr: %s",
+            self.ip_address,
+            self.serial_number,
+        )
+
+        placeholders = {
+            CONF_HOST: self.ip_address,
+            WALLCONNECTOR_SERIAL_NUMBER: self.serial_number,
+        }
+
+        self.context["title_placeholders"] = placeholders
+        return await self.async_step_user()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        data_schema = vol.Schema(
+            {vol.Required(CONF_HOST, default=self.ip_address): str}
+        )
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=data_schema)
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except WallConnectorError:
+            errors["base"] = "cannot_connect"
+        except Exception as ex:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception: %s", ex)
+            errors["base"] = "unknown"
+
+        if not errors:
+            existing_entry = await self.async_set_unique_id(
+                info[WALLCONNECTOR_SERIAL_NUMBER]
+            )
+            if existing_entry:
+                self.hass.config_entries.async_update_entry(
+                    existing_entry, data=user_input
+                )
+                await self.hass.config_entries.async_reload(existing_entry.entry_id)
+                return self.async_abort(reason="already_configured")
+
+            return self.async_create_entry(title=info["title"], data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for Tesla Wall Connector."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Handle options flow."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        data_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_SCAN_INTERVAL,
```

Comment:
We don't want to allow configuring scan interval. All integrations with config entries support allow the user to disable automatic polling and the user can then automate their own updates with the entity service `homeassistant.update_entity`.

Please remove this option.